x-pack/plugins/ingest_pipelines/public/application/components/pipeline_form/pipeline_test_flyout/pipeline_test_flyout_provider.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useEffect } from 'react';
+
+import { Pipeline } from '../../../../../common/types';
+import { useFormContext } from '../../../../shared_imports';
+import { PipelineTestFlyout, PipelineTestFlyoutProps } from './pipeline_test_flyout';
+
+type Props = Omit<PipelineTestFlyoutProps, 'pipeline' | 'isPipelineValid'>;
+
+export const PipelineTestFlyoutProvider: React.FunctionComponent<Props> = ({
```

Comment:
Just a q here; why have we separated the setting up of `useForm` and `PipelineTestFlyout` by adding a `PipelineTestFlyoutProvider`? I would have expected `PipelineTestFlyout` to be memoized or similar to benefit from this pattern. Not saying this needs to change, just curious!

Reply:
TBH I was following a similar pattern to what Seb proposed here for the request flyout: https://github.com/alisonelizabeth/kibana/pull/2. I'm going to leave as is for now to try to get this PR in before the demo today, but happy to address in a separate PR. Let me know what you think.